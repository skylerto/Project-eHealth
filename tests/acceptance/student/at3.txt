add_physician ( 1,"Mayo",specialist )
add_physician ( 2,"Mayo",specialist )
add_physician ( 3,"Mayo",generalist )
add_patient ( 3,"Drew" )
add_patient ( 1,"Drew" )
add_medication ( 1,["Wafarin",pill,1.0,6.0] )
add_medication ( 2,["acetaminophen",liquid,1.0,25.5] )
new_prescription ( 2,1,3 )
new_prescription ( 4,3,1 )
new_prescription ( 1,1,1 )
new_prescription ( 3,2,3 )
add_medicine(1,1,5.5)
add_medicine(1,2,5.5)
add_medicine(4,2,5.5)
add_medicine(2,2,5.5)
add_medicine(2,1,5.5)
add_medicine(3,1,5.5)
add_physician(-1,"Mayo",specialist)
add_physician(1,"Mayo",specialist)
add_physician(4,"99ayo",specialist)
prescriptions_q(-1000)
prescriptions_q(1)
prescriptions_q(2)
prescriptions_q(5)
add_interaction(1,2)
add_medicine(4,1,5.5)
