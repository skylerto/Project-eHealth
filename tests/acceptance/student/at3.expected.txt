  0: ok
  Physicians:
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(1,"Mayo",specialist)
  1: ok
  Physicians:
    1->[Mayo,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(2,"Mayo",specialist)
  2: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_physician(3,"Mayo",generalist)
  3: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(3,"Drew")
  4: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    3->Drew
  Medications: 
  Interactions: 
  Prescriptions:
->add_patient(1,"Drew")
  5: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
  Interactions: 
  Prescriptions:
->add_medication(1,["Wafarin", pill, 1, 6])
  6: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
  Interactions: 
  Prescriptions:
->add_medication(2,["acetaminophen", liquid, 1, 25.5])
  7: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
->new_prescription(2,1,3)
  8: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    2->[2,1,3,()]
->new_prescription(4,3,1)
  9: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    2->[2,1,3,()]
    4->[4,3,1,()]
->new_prescription(1,1,1)
  10: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
    4->[4,3,1,()]
->new_prescription(3,2,3)
  11: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,()]
    2->[2,1,3,()]
    3->[3,2,3,()]
    4->[4,3,1,()]
->add_medicine(1,1,5.5)
  12: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50)]
    2->[2,1,3,()]
    3->[3,2,3,()]
    4->[4,3,1,()]
->add_medicine(1,2,5.5)
  13: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,()]
    3->[3,2,3,()]
    4->[4,3,1,()]
->add_medicine(4,2,5.5)
  14: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,()]
    3->[3,2,3,()]
    4->[4,3,1,(2->5.50)]
->add_medicine(2,2,5.5)
  15: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50)]
    3->[3,2,3,()]
    4->[4,3,1,(2->5.50)]
->add_medicine(2,1,5.5)
  16: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,()]
    4->[4,3,1,(2->5.50)]
->add_medicine(3,1,5.5)
  17: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,(1->5.50)]
    4->[4,3,1,(2->5.50)]
->add_physician(-1,"Mayo",specialist)
  18: physician id must be a positive integer
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,(1->5.50)]
    4->[4,3,1,(2->5.50)]
->add_physician(1,"Mayo",specialist)
  19: physician id already in use
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,(1->5.50)]
    4->[4,3,1,(2->5.50)]
->add_physician(4,"99ayo",specialist)
  20: name must start with a letter
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,(1->5.50)]
    4->[4,3,1,(2->5.50)]
->prescriptions_q(-1000)
  21: medication id must be a positive integer
->prescriptions_q(1)
  22: ok
  Output: medication is Wafarin
    1->Drew
    3->Drew
->prescriptions_q(2)
  23: ok
  Output: medication is acetaminophen
    1->Drew
    3->Drew
->prescriptions_q(5)
  24: medication id must be registered
->add_interaction(1,2)
  25: ok
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,(1->5.50)]
    4->[4,3,1,(2->5.50)]
->add_medicine(4,1,5.5)
  26: specialist is required to add a dangerous interaction
  Physicians:
    1->[Mayo,sp]
    2->[Mayo,sp]
    3->[Mayo,gn]
  Patients: 
    1->Drew
    3->Drew
  Medications: 
    1->[Wafarin,pl,1.00,6.00]
    2->[acetaminophen,lq,1.00,25.50]
  Interactions: 
    [Wafarin,pl,1]->[acetaminophen,lq,2]
  Prescriptions:
    1->[1,1,1,(1->5.50,2->5.50)]
    2->[2,1,3,(2->5.50,1->5.50)]
    3->[3,2,3,(1->5.50)]
    4->[4,3,1,(2->5.50)]
